main.js

```diff
@@ -769,17 +893,56 @@ function updateIniFromForm(ini) {
     }
 }
 
-function updateSelection(id, value, list) {
+function updateSelection(id, value, newChoiceArray) {
     "use strict";
-    value = parseInt(value.trim());
-    while (document.getElementById(id).firstChild) {
-        document.getElementById(id).removeChild(document.getElementById(id).firstChild);
+    var select = document.getElementById(id);
+    
+    // Sanitize value
+    if (typeof value === "string") {
+        value = parseInt(value.trim());
```

Comment:
```suggestion
        value = Number(value.trim());
```
Probably in other places where `parseInt` is being used, too, unless you're using it to parse non-decimal numbers.